File: bindings/node/nvs.js
// NaveScript Node.js Binding
const addon = require('./build/Release/nvs_runtime.node');

class NvsRuntime {
    constructor() {
        this.runtime = addon.create_runtime();
    }
    
    run(code) {
        return addon.run(this.runtime, code);
    }
}

module.exports = NvsRuntime;

File: bindings/ruby/nvs.rb
# NaveScript Ruby Binding
require 'ffi'

module Nvs
  extend FFI::Library
  ffi_lib 'nvs_runtime'
  
  attach_function :nvs_create_runtime, [], :pointer
  attach_function :nvs_run, [:pointer, :string], :void
  
  class Runtime
    def initialize
      @handle = Nvs.nvs_create_runtime
    end
    
    def run(code)
      Nvs.nvs_run(@handle, code)
    end
  end
end
